lua/nvim-treesitter/textobjects/attach.lua

```diff
@@ -26,9 +27,23 @@ function M.make_attach(normal_mode_functions, submodule, keymap_modes)
           mapping_description = function_description .. " " .. query_metadata
         end
 
-        vim.keymap.set(keymap_modes, mapping, function()
-          require("nvim-treesitter.textobjects." .. submodule)[function_call](query)
-        end, { buffer = bufnr, silent = true, remap = false, desc = mapping_description })
+        local function fn()
+          if opts.repeatable then
+            _G._nvim_treesitter_textobject_last_function = function()
+              require("nvim-treesitter.textobjects." .. submodule)[function_call](query)
+            end
+            vim.o.opfunc = "v:lua._nvim_treesitter_textobject_last_function"
+            vim.api.nvim_feedkeys("g@l", "n", false)
```

Comment:
going via `opfunc` is also a bit of a hack

Reply:
need to check for which modes this can work

Reply:
If I understand correctly, the `opfunc` doesn't accept lua function, so you hacked the function to be a vimscript-like function and used it?
And you're not sure if the repeatable function works with select move etc?

Reply:
yes, I made it a vim function through a global. https://github.com/lewis6991/gitsigns.nvim/blob/main/lua/gitsigns/repeat.lua does not work because we would need to write `require'nvim-treesitter.textobject'...` and Vim does not parse the opfunc when there is a `-` in the string.

I don't know for which other functions `.` could be applicable. Typically `.` is to repeat edits. Swap is the only module that performs edits. There was a desire to introduce `;` for repeating the motions but this would be a different functionality and PR.

Reply:
> There was a desire to introduce ; for repeating the motions but this would be a different functionality and PR.

That would be nice! At the moment I kind of hack it through "dahu/vim-fanfingtastic" and  "Houl/repmo-vim".